Changes needed to use Studio libsunperf on Solaris (if available).

These changes will be sent upstream.

--- gsl-2.4/configure.ac	2018-01-05 17:41:59.207892483 +0000
+++ gsl-2.4/configure.ac	2018-01-05 17:41:37.280831214 +0000
@@ -125,6 +125,21 @@ AC_SUBST(GSL_CFLAGS)
 AC_SUBST(GSL_LIBS)
 AC_SUBST(GSL_LIBM)
 
+dnl Check for existing BLAS
+dnl look for BLAS in some typical places
+AM_CONDITIONAL(SOLARIS_SUNPERF, false)
+case "$host" in
+*-*-solaris*) 
+    AC_CHECK_LIB(sunperf,dcopy,BLAS=-lsunperf)
+    if test "$BLAS"; then
+        dnl Solaris BLAS library has performance benefits
+        AM_CONDITIONAL(SOLARIS_SUNPERF, true)
+        CFLAGS="$CFLAGS -xlibmopt -xbuiltin=%default -xlibmil"
+        GSL_LDFLAGS="$GSL_LDFLAGS -lsunperf"
+    fi
+    ;;
+esac
+
 if test "$ac_cv_c_inline" != no ; then 
 dnl Check for "extern inline", using a modified version of the test
 dnl for AC_C_INLINE from acspecific.mt
--- gsl-2.2.1/Makefile.am.orig	2016-11-28 09:04:30.000000000 +0000
+++ gsl-2.2.1/Makefile.am	2016-11-30 08:35:26.586505711 +0000
@@ -2,7 +2,14 @@
 
 # AUTOMAKE_OPTIONS = readme-alpha
 
-SUBDIRS = gsl utils sys test err const complex cheb block vector matrix permutation combination multiset sort ieee-utils cblas blas linalg eigen specfunc dht qrng rng randist fft poly fit multifit multifit_nlinear multilarge multilarge_nlinear rstat statistics siman sum integration interpolation histogram ode-initval ode-initval2 roots multiroots min multimin monte ntuple diff deriv cdf wavelet bspline spblas spmatrix splinalg doc
+SUBDIRS = gsl utils sys test err const complex cheb block vector matrix permutation combination multiset sort ieee-utils $(CBLAS_DIRS) blas linalg eigen specfunc dht qrng rng randist fft poly fit multifit multifit_nlinear multilarge multilarge_nlinear rstat statistics siman sum integration interpolation histogram ode-initval ode-initval2 roots multiroots min multimin monte ntuple diff deriv cdf wavelet bspline spblas spmatrix splinalg doc
+
+SOLARIS_SUNPERF = @SOLARIS_SUNPERF@
+if SOLARIS_SUNPERF
+    CBLAS_DIRS=
+else
+    CBLAS_DIRS=cblas
+endif
 
 SUBLIBS = block/libgslblock.la blas/libgslblas.la bspline/libgslbspline.la complex/libgslcomplex.la cheb/libgslcheb.la dht/libgsldht.la diff/libgsldiff.la deriv/libgslderiv.la eigen/libgsleigen.la err/libgslerr.la fft/libgslfft.la fit/libgslfit.la histogram/libgslhistogram.la ieee-utils/libgslieeeutils.la integration/libgslintegration.la interpolation/libgslinterpolation.la linalg/libgsllinalg.la matrix/libgslmatrix.la min/libgslmin.la monte/libgslmonte.la multifit/libgslmultifit.la multifit_nlinear/libgslmultifit_nlinear.la multilarge/libgslmultilarge.la multilarge_nlinear/libgslmultilarge_nlinear.la multimin/libgslmultimin.la multiroots/libgslmultiroots.la ntuple/libgslntuple.la ode-initval/libgslodeiv.la ode-initval2/libgslodeiv2.la permutation/libgslpermutation.la combination/libgslcombination.la multiset/libgslmultiset.la poly/libgslpoly.la qrng/libgslqrng.la randist/libgslrandist.la rng/libgslrng.la roots/libgslroots.la siman/libgslsiman.la sort/libgslsort.la specfunc/libgslspecfunc.la rstat/libgslrstat.la statistics/libgslstatistics.la sum/libgslsum.la sys/libgslsys.la test/libgsltest.la utils/libutils.la vector/libgslvector.la cdf/libgslcdf.la wavelet/libgslwavelet.la spmatrix/libgslspmatrix.la spblas/libgslspblas.la splinalg/libgslsplinalg.la
 
@@ -29,10 +36,18 @@
 bin_PROGRAMS = gsl-randist gsl-histogram
 
 gsl_randist_SOURCES = gsl-randist.c
+if SOLARIS_SUNPERF
+gsl_randist_LDADD = libgsl.la
+else
 gsl_randist_LDADD = libgsl.la cblas/libgslcblas.la
+endif
 
 gsl_histogram_SOURCES = gsl-histogram.c
+if SOLARIS_SUNPERF
+gsl_histogram_LDADD = libgsl.la
+else
 gsl_histogram_LDADD = libgsl.la cblas/libgslcblas.la
+endif
 
 check_SCRIPTS = test_gsl_histogram.sh pkgconfig.test
 TESTS = test_gsl_histogram.sh pkgconfig.test
